### PR TITLE
An improved and configurable Slideshow

### PR DESCRIPTION
**Pull request by mxdvl**
## What does this change?

- Add a negative delay for the first image
- Remove the figure caption gradient if there is no caption
- Refactor the component so the configuration makes sense
- Handle the case of a single image being passed down
- Add some stories for the `Slideshow` component

## Why?

Currently, the slideshow fades in, which looks odd. When images cross-fade, it’s possible to see the background through as they both have some transparency.

Help empower the current maintainers to feel empowered to make relevant changes to this component. Make it more visually pleasing. Improve [on Frontend’s version](https://github.com/guardian/frontend/pull/10325).

Crossfade is [hard](https://stackoverflow.com/a/73874280/1577447). I [made a visualisation to explain this further](https://www.mxdvl.com/tools/slideshow).

screenshot of slideshow visualisation

## Screenshots

**Before** (notice initial white background)

https://github.com/guardian/dotcom-rendering/assets/76776/62781b4f-4bf8-4e48-a6a3-047d98fdb669

**After** (smooth as)

https://github.com/guardian/dotcom-rendering/assets/76776/7255c1d5-b5d4-492e-9590-50b0026f3ea4

### CODE REVIEW

File: dotcom-rendering/src/components/Slideshow.tsx [hunk 1]
```diff
@@ -1,61 +1,59 @@
...
 	// Generate and return a keyframes name for this animation
 	return keyframes`
-		${stageOne} {opacity: 0;}
-		${stageTwo} {opacity: 1;}
-		${stageThree} {opacity: 1;}
-		${stageFour} {opacity: 0;}
+		${`${stages[0] / totalLoopTime}%`} {opacity: 0;}
+		${`${stages[1] / totalLoopTime}%`} {opacity: 1;}
+		${`${stages[2] / totalLoopTime}%`} {opacity: 1;}
+		${`${stages[3] / totalLoopTime}%`} {opacity: 0;}
```
Comment (JamieB-gu) [inline dotcom-rendering/src/components/Slideshow.tsx:40]: I don't think this is a necessary change, but it looks like there might be a coupling between the stage and the opacity so would it make sense to capture them together like this? Happy for you to tell me this doesn't make sense!

```ts
const stages = [
	{ time: 0, opacity: 0 },
	{ time: FADING_DURATION, opacity: 1 },
	{ time: FADING_DURATION + DISPLAY_DURATION + FADING_DURATION, opacity: 1 },
	{ time: FADING_DURATION + DISPLAY_DURATION + FADING_DURATION + FADING_DURATION, opacity: 0 },
];

return keyframes(stages.map(({ time, opacity }) => `${time / totalLoopTime}% { opacity: ${opacity}; }`));
```

**Review by JamieB-gu — APPROVED** [review]
Looks great and easier to understand, thanks for doing this!

Couple of suggestions but nothing blocking.

File: dotcom-rendering/src/components/Slideshow.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+	</Wrapper>
+);
+export const WithTwoImages = () => (
+	<Wrapper>
+		<Slideshow images={images.slice(0, 2)} imageSize="large" />
+	</Wrapper>
+);
+export const WithFiveImages = () => (
+	<Wrapper>
+		<Slideshow images={images} imageSize="large" />
```
[the remark this answers is not in the document]
Comment (mxdvl) [reply dotcom-rendering/src/components/Slideshow.stories.tsx:61]: Good shout, not sure I’ll have the time to address as part of this PR, but will try :)

File: dotcom-rendering/src/components/Slideshow.tsx [hunk 3]
```diff
@@ -136,96 +89,115 @@ const captionStyles = css`
...
+ *   image 3 : ▁▁▁▁▁▁▁▁▁▁▁▁▃▅███████▁▁▁▁▁▁▁ (fades in, off after 4 is fully opaque)
  *
- *                 time ->
- *   image-1 > ####|----|----|----
- *   image-2 > ----|####|----|----
- *   image-3 > ----|----|####|----
- *   image-4 > ----|----|----|####
+ *   image 4 : ▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▃▅█████▅▃ (fades in and out)
+ *       time →     →     →     →     →
+ *  ```
```
Comment (AshCorr) [inline dotcom-rendering/src/components/Slideshow.tsx:130]: Really cool description!

**Review by AshCorr — APPROVED** [review]
Nice!

File: dotcom-rendering/src/components/Slideshow.tsx (hunk 3 above)
Comment (mxdvl) [reply dotcom-rendering/src/components/Slideshow.tsx:130]: Graphic Design is My Passion

**Pull request merged**